subgraphs/compound-forks/src/mapping.ts

```diff
@@ -0,0 +1,1445 @@
+      market.cumulativeLiquidateUSD
+    );
+    cumulativeTotalRevenueUSD = cumulativeTotalRevenueUSD.plus(
+      market._cumulativeTotalRevenueUSD
+    );
+    cumulativeProtocolSideRevenueUSD = cumulativeProtocolSideRevenueUSD.plus(
+      market._cumulativeProtocolSideRevenueUSD
+    );
+    cumulativeSupplySideRevenueUSD = cumulativeSupplySideRevenueUSD.plus(
+      market._cumulativeSupplySideRevenueUSD
+    );
+  }
+  protocol.totalValueLockedUSD = totalValueLockedUSD;
+  protocol.totalDepositBalanceUSD = totalDepositBalanceUSD;
+  protocol.totalBorrowBalanceUSD = totalBorrowBalanceUSD;
+  protocol.cumulativeBorrowUSD = cumulativeBorrowUSD;
+  protocol.cumulativeDepositUSD = cumulativeDepositUSD;
+  protocol.cumulativeLiquidateUSD = cumulativeLiquidateUSD;
+  protocol.cumulativeTotalRevenueUSD = cumulativeTotalRevenueUSD;
+  protocol.cumulativeProtocolSideRevenueUSD = cumulativeProtocolSideRevenueUSD;
+  protocol.cumulativeSupplySideRevenueUSD = cumulativeSupplySideRevenueUSD;
+  protocol.save();
+}
+
+function setSupplyInterestRate(marketID: string, rate: BigDecimal): void {
+  setInterestRate(marketID, rate, true);
+}
+
+function setBorrowInterestRate(marketID: string, rate: BigDecimal): void {
+  setInterestRate(marketID, rate, false);
+}
+
+function setInterestRate(
+  marketID: string,
+  rate: BigDecimal,
+  isSupply: boolean
+): void {
+  let market = Market.load(marketID);
+  if (!market) {
+    log.warning("[setInterestRate] Market not found: {}", [marketID]);
+    return;
+  }
+  if (market.rates.length < 2) {
+    log.warning("[setInterestRate] Market has less than 2 rates: {}", [
+      marketID,
+    ]);
+    return;
+  }
+  let supplyInterestRateID = market.rates[0];
+  let borrowInterestRateID = market.rates[1];
+  let supplyInterestRate = InterestRate.load(supplyInterestRateID);
+  if (!supplyInterestRate) {
+    log.warning("[setInterestRate] Supply interest rate not found: {}", [
+      supplyInterestRateID,
+    ]);
+    return;
+  }
+  let borrowInterestRate = InterestRate.load(borrowInterestRateID);
+  if (!borrowInterestRate) {
+    log.warning("[setInterestRate] Borrow interest rate not found: {}", [
+      borrowInterestRateID,
+    ]);
+    return;
+  }
+  if (isSupply) {
+    supplyInterestRate.rate = rate;
+    supplyInterestRate.save();
+  } else {
+    borrowInterestRate.rate = rate;
+    borrowInterestRate.save();
+  }
+  market.rates = [supplyInterestRateID, borrowInterestRateID];
+  market.save();
+}
+
+function getTokenPriceUSD(
+  getUnderlyingPriceResult: ethereum.CallResult<BigInt>,
+  underlyingDecimals: i32
+): BigDecimal {
+  let mantissaDecimalFactor = 18 - underlyingDecimals + 18;
+  let bdFactor = exponentToBigDecimal(mantissaDecimalFactor);
+  return getOrElse<BigInt>(getUnderlyingPriceResult, BIGINT_ZERO)
+    .toBigDecimal()
+    .div(bdFactor);
+}
+
+function getMarketHourlySnapshotID(marketID: string, timestamp: i32): string {
+  return marketID
```

Comment:
This is actually changed in schema 1.2.1

Reply:
updated to 1.2.1